File: implementation/scipamato/core/web/src/test/java/ch/difty/scipamato/core/web/resources/jasper/JasperReportExceptionTest.java
package ch.difty.scipamato.core.web.resources.jasper;

import static org.assertj.core.api.Assertions.assertThat;

import net.sf.jasperreports.engine.JRException;
import org.apache.wicket.util.resource.ResourceStreamNotFoundException;
import org.junit.jupiter.api.Test;

class JasperReportExceptionTest {

    private JasperReportException exception;

    @Test
    void makeJasperReportExceptionWithJRException() {
        final JRException e = new JRException("foo");
        exception = new JasperReportException(e);
        assertThat(exception.getMessage()).isEqualTo("net.sf.jasperreports.engine.JRException: foo");
    }

    @Test
    void makeJasperReportExceptionWithResourceStreamNotFoundException() {
        final ResourceStreamNotFoundException e = new ResourceStreamNotFoundException("foo");
        exception = new JasperReportException(e);
        assertThat(exception.getMessage()).isEqualTo(
            "org.apache.wicket.util.resource.ResourceStreamNotFoundException: foo");
    }

    @Test
    void makeJasperReportException_withStringMessage() {
        exception = new JasperReportException("foo");
        assertThat(exception.getMessage()).isEqualTo("foo");
    }
}

File: core/core-web/src/test/kotlin/ch/difty/scipamato/core/web/resources/jasper/JasperReportExceptionTest.kt
package ch.difty.scipamato.core.web.resources.jasper

import net.sf.jasperreports.engine.JRException
import org.amshove.kluent.shouldBeEqualTo
import org.apache.wicket.util.resource.ResourceStreamNotFoundException
import org.junit.jupiter.api.Test

internal class JasperReportExceptionTest {

    private lateinit var exception: JasperReportException

    @Test
    fun makeJasperReportExceptionWithJRException() {
        val e = JRException("foo")
        exception = JasperReportException(e)
        exception.message shouldBeEqualTo "net.sf.jasperreports.engine.JRException: foo"
    }

    @Test
    fun makeJasperReportExceptionWithResourceStreamNotFoundException() {
        val e = ResourceStreamNotFoundException("foo")
        exception = JasperReportException(e)
        exception.message shouldBeEqualTo "org.apache.wicket.util.resource.ResourceStreamNotFoundException: foo"
    }

    @Test
    fun makeJasperReportException_withStringMessage() {
        exception = JasperReportException("foo")
        exception.message shouldBeEqualTo "foo"
    }
}
